src/test/java/com/gui/HomePageTest.java

```diff
@@ -5,19 +5,44 @@
 import com.gui.pages.desktop.ProductsPage;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.Assert;
 import org.testng.annotations.Test;
 
 
-public class HomePageTest  {
-    WebDriver driver = new ChromeDriver();
+public class HomePageTest extends AbstractTest {
 
     @Test
     public void test(){
-        driver.get("https://automationteststore.com/");
         HomePage homePage =new HomePage(driver);
         ProductsPage productsPage= homePage.searchKeyWord("shirt");
         productsPage.printAllItems();
     }
 
+    @Test(groups = "group1")
+    public void testCurrencyChange() {
+        HomePage homePage =new HomePage(driver);
+        homePage.clickCurrencySelector();
+        homePage.clickItemInCurrencySelector(0); //chooses the pound currency
+        Assert.assertTrue(homePage.isCurrencyInCartCorrect("£"));
+    }
+
+    @Test(groups = "group1")
+    public void testCartItemCount(){
+        HomePage homePage =new HomePage(driver);
+        homePage.clickAddToCartButton();
+        homePage.clickAddToCartButton();
+        homePage.clickAddToCartButton();
+        Assert.assertTrue(homePage.isCartItemCountCorrect("3"));
+    }
+
+    @Test(groups = "group1")
+    public void testCartTotal(){
+        HomePage homePage =new HomePage(driver);
+        homePage.clickAddToCartButton();
+        Assert.assertTrue(homePage.isCartTotalCorrect("$29.50"));
```

Comment:
If the value "$29.50" changes due to logic or product price changes, the test will break unnecessarily. Consider parameterizing it or verifying the presence of a symbol + numeric pattern if exact price isn't critical.